File: day2.py
#!/usr/bin/env python3

with open("day2.txt") as f:
    input = f.read().split("\n")

horizontal = 0
depth = 0
aim = 0

for line in input[:-1]:
    value = int(line.split()[-1])
    if "forward" in line:
        horizontal += value
        depth += aim * value
    elif "down" in line:
        aim += value
    elif "up" in line:
        aim -= value
print("Part 1")
print(horizontal * aim)
print("Part 2")
print(horizontal * depth)

File: day2.cpp
#include <fstream>
#include <string>
#include <iostream>
#include <sstream>

int main() {
  std::ifstream infile("day2.txt");
  std::string line;
  int horizontal {0};
  int depth {0};
  int aim {0};
  while (std::getline(infile, line)) {
    std::istringstream iss(line);
    std::string command;
    int value;

    if(!(iss >> command >> value)) { break;}
    if (command == "forward"){
      horizontal += value;
      depth += aim * value;
    } else if (command == "down") {
      aim += value;
    } else if (command == "up") {
      aim -= value;
    }
  }

  std::cout << "Part 1: " << horizontal * aim << "\n";
  std::cout << "Part 2: " << horizontal * depth << "\n";
}
